tests/cpp_api/audio_tests/audio_tests.cpp

```diff
@@ -217,22 +304,32 @@ int test(int test_case, const char *path, int qa_mode, int downmix, int gpu) {
         int *label_id = reinterpret_cast<int *>(labels->at(0)->buffer());  // The labels are present contiguously in memory
         std::cout << "Audio file : " << audio_file_name << "\n";
         std::cout << "Label : " << *label_id << "\n";
-        for (uint idx = 0; idx < output_tensor_list->size(); idx++) {
-            buffer = static_cast<float *>(output_tensor_list->at(idx)->buffer());
-            output_tensor_list->at(idx)->copy_roi(roi.data());
-            max_channels = output_tensor_list->at(idx)->dims().at(2);
-            max_samples = max_channels == 1 ? 1 : output_tensor_list->at(idx)->dims().at(1);
-            frames = roi[idx * 4 + 2];
-            channels = roi[idx * 4 + 3];
-            buffer_size = roi[idx * 4 + 2] * roi[idx * 4 + 3];
+        if(test_case == 8){
+            nsr_begin = static_cast<int *>(output_tensor_list->at(0)->buffer());
+            nsr_length = static_cast<int *>(output_tensor_list->at(1)->buffer());
+        }
+        else{
+            for (uint idx = 0; idx < output_tensor_list->size(); idx++) {
+                buffer = static_cast<float *>(output_tensor_list->at(idx)->buffer());
+                output_tensor_list->at(idx)->copy_roi(roi.data());
+                max_channels = output_tensor_list->at(idx)->dims().at(2);
+                max_samples = max_channels == 1 ? 1 : output_tensor_list->at(idx)->dims().at(1);
+                frames = roi[idx * 4 + 2];
+                channels = roi[idx * 4 + 3];
+                buffer_size = roi[idx * 4 + 2] * roi[idx * 4 + 3];
+            }
         }
     }
 
     if (qa_mode) {
         std::cout << "\n *****************************Verifying Audio output**********************************\n";
         if (verify_output(buffer, frames, channels, case_name, max_samples, max_channels, buffer_size)) {
             std::cout << "PASSED!\n\n";
-        } else {
+        }
+        else if (test_case == 8 && verify_non_silent_region_output(nsr_begin, nsr_length, case_name)) {
```

Comment:
can we not club it into the verify_output function and pass a test_case there? why do we need separate function for this?

Reply:
The verify NSR function is added because it needs two arguments not present in the verify_output function